internal/controllers/machine/machine_controller_status.go

```diff
@@ -499,6 +586,49 @@ func setReadyCondition(ctx context.Context, machine *clusterv1.Machine) {
 	v1beta2conditions.Set(machine, *readyCondition)
 }
 
+// calculateDeletingConditionForSummary calculates a Deleting condition for the calculation of the Ready condition
+// (which is done via a summary). This is necessary to avoid including the verbose details of the Deleting condition
+// message in the summary.
+// This is important to ensure we have a limited amount of unique messages across Machines. This allows us to deduplicate
+// messages when aggregating Ready conditions of many Machines into the MachinesReady condition of e.g. the MachineSet.
+// For the same reason we are only surfacing messages with "more than 30m" instead of using the exact durations.
+// We assume that 30m is a duration after which it makes sense to notify users that Node drains and waiting for volume
+// detach are still in progress.
+func calculateDeletingConditionForSummary(machine *clusterv1.Machine) v1beta2conditions.ConditionWithOwnerInfo {
```

Comment:
nit
```suggestion
// calculateDeletingConditionForSummary calculates a Deleting condition for the calculation of the Ready condition
// (which is done via a summary). This is necessary to avoid including the verbose details of the Deleting condition
// message in the summary.
// This is also important to ensure we have a limited amount of unique messages across Machines thus allowing to
// nicely aggregate Ready conditions from many Machines into the MachinesReady condition of e.g. the MachineSet.
// For the same reason we are only surfacing messages with "more than 30m" instead of using the exact durations.
// "30m" is a an arbitrary duration after which we assume it makes sense to give more emphasis on 
the fact that Node drains and waiting for volume
// detach are still in progress. 
func computeDeletingOverrideConditionForSummary(machine *clusterv1.Machine) v1beta2conditions.ConditionWithOwnerInfo {
```

Reply:
~ changed to this. I didn't use arbitrary because it would mean we randomly picked this value, but we chose it intentionally.